Type error in emitter.d.ts

I'm getting an error when trying to build Quill2.0-beta.0 with an Angular 17 project using Webpack or esbuild:

**Actual behavior**:
```
Error: node_modules/quill/core/emitter.d.ts:23:5 - error TS2416: Property 'listeners' in type 'Emitter' is not assignable to the same property in base type 'EventEmitter<string, any>'.
  Type 'Record<string, { node: Node; handler: Function; }[]>' is not assignable to type '<T extends string>(event: T) => ((...args: any[]) => void)[]'.
    Type 'Record<string, { node: Node; handler: Function; }[]>' provides no match for the signature '<T extends string>(event: T): ((...args: any[]) => void)[]'.

23     listeners: Record<string, {
```

**Platforms**:
Webpack or esbuild

**Version**:
v2.0.0-beta.0

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,7 @@
 # [Unreleased]
 
 - Fix syntax label from "Javascript" to "JavaScript".
+- Fix typing errors for emitter.
 
 # v2.0.0-beta.0
 

--- a/packages/quill/package.json
+++ b/packages/quill/package.json
@@ -6,7 +6,7 @@
   "homepage": "https://quilljs.com",
   "main": "quill.js",
   "dependencies": {
-    "eventemitter3": "^4.0.7",
+    "eventemitter3": "^5.0.1",
     "lodash-es": "^4.17.21",
     "parchment": "^3.0.0-alpha.1",
     "quill-delta": "^5.1.0"
@@ -23,7 +23,7 @@
     "@types/webpack": "^5.28.5",
     "@typescript-eslint/eslint-plugin": "^6.12.0",
     "@typescript-eslint/parser": "^6.12.0",
-    "@vitest/browser": "^1.0.0-beta.5",
+    "@vitest/browser": "^1.1.3",
     "babel-loader": "^9.1.3",
     "babel-plugin-transform-define": "^2.1.4",
     "css-loader": "^6.8.1",
@@ -47,7 +47,7 @@
     "ts-loader": "^9.5.1",
     "ts-node": "^10.9.1",
     "typescript": "5.2.2",
-    "vitest": "^1.0.0-beta.5",
+    "vitest": "^1.1.3",
     "webpack": "^5.89.0",
     "webpack-cli": "^5.1.4",
     "webpack-dev-server": "^4.15.1",

--- a/packages/quill/src/core/emitter.ts
+++ b/packages/quill/src/core/emitter.ts
@@ -39,12 +39,11 @@ class Emitter extends EventEmitter<string> {
     USER: 'user',
   } as const;
 
-  // @ts-expect-error listeners is declared in EventEmitter as a function
-  listeners: Record<string, { node: Node; handler: Function }[]>;
+  protected domListeners: Record<string, { node: Node; handler: Function }[]>;
 
   constructor() {
     super();
-    this.listeners = {};
+    this.domListeners = {};
     this.on('error', debug.error);
   }
 
@@ -55,7 +54,7 @@ class Emitter extends EventEmitter<string> {
   }
 
   handleDOM(event: Event, ...args: unknown[]) {
-    (this.listeners[event.type] || []).forEach(({ node, handler }) => {
+    (this.domListeners[event.type] || []).forEach(({ node, handler }) => {
       if (event.target === node || node.contains(event.target as Node)) {
         handler(event, ...args);
       }
@@ -63,10 +62,10 @@ class Emitter extends EventEmitter<string> {
   }
 
   listenDOM(eventName: string, node: Node, handler: EventListener) {
-    if (!this.listeners[eventName]) {
-      this.listeners[eventName] = [];
+    if (!this.domListeners[eventName]) {
+      this.domListeners[eventName] = [];
     }
-    this.listeners[eventName].push({ node, handler });
+    this.domListeners[eventName].push({ node, handler });
   }
 }
 

--- a/packages/quill/test/unit/core/emitter.spec.ts
+++ b/packages/quill/test/unit/core/emitter.spec.ts
@@ -1,0 +1,38 @@
+import { describe, expect, test } from 'vitest';
+import Emitter from '../../../src/core/emitter';
+import Quill from '../../../src/core';
+
+describe('emitter', () => {
+  test('emit and on', () => {
+    const emitter = new Emitter();
+
+    let received: unknown;
+    emitter.on('abc', (data) => {
+      received = data;
+    });
+    emitter.emit('abc', { hello: 'world' });
+
+    expect(received).toEqual({ hello: 'world' });
+  });
+
+  test('listenDOM', () => {
+    const quill = new Quill(document.createElement('div'));
+    document.body.appendChild(quill.container);
+
+    let calls = 0;
+    quill.emitter.listenDOM('click', document.body, () => {
+      calls += 1;
+    });
+
+    document.body.click();
+    expect(calls).toEqual(1);
+
+    quill.container.remove();
+    document.body.click();
+    expect(calls).toEqual(1);
+
+    document.body.appendChild(quill.container);
+    document.body.click();
+    expect(calls).toEqual(2);
+  });
+});

